src/tests/ftest/process_cores.py

```diff
@@ -0,0 +1,454 @@
+
+
+def install_packages(package_list, args):
+    """Install any packages required to generate a stacktrace.
+
+    Args:
+        package_list (list): list of package names from which to install debuginfo packages
+        args (argparse.Namespace): command line arguments for this program
+
+    Returns:
+        bool: True if all the packages installed successfully; False otherwise
+
+    """
+    # The list of commands - as a list of space-separated parts of each command
+    command_list = []
+
+    # Remove any "source tree" test hackery that might interfere with RPM installation
+    path = os.path.join(os.path.sep, "usr", "share", "spdk", "include")
+    if os.path.islink(path):
+        command_list.append(["sudo", "rm", "-f", path])
+
+    # Add installing any required packages
+    distro_info = detect()
+    install_packages = PACKAGE_NAMES.copy()
+    install_options = ["--exclude", "ompi-debuginfo"]
+    if os.getenv("TEST_RPMS", 'false') == 'true':
+        if "suse" in distro_info.name.lower():
+            install_packages.extend(["libpmemobj1", "python3", "openmpi3"])
+        elif "centos" in distro_info.name.lower() and distro_info.version == "7":
+            install_options.extend(["--enablerepo=*-debuginfo", "--exclude", "nvml-debuginfo"])
+            install_packages.extend(["libpmemobj", "python36", "openmpi3", "gcc"])
+        elif is_el(distro_info) and distro_info.version == "8":
+            install_options.append("--enablerepo=*-debuginfo")
+            install_packages.extend(["libpmemobj", "python3", "openmpi", "gcc"])
+        else:
+            raise RuntimeError("Error unsupported distro: {}".format(distro_info))
+    command_list.append(["sudo", "dnf", "-y", "install"] + install_options + install_packages)
+
+    # Add installing any debuginfo packages
+    debuginfo_packages = []
+    if is_el(distro_info):
+        debuginfo_packages.append(PackageInfo("python3"))
+    for name in package_list:
+        info = get_package_info(name, args)
+        if info is not None:
+            debuginfo_packages.append(str(info))
+    if debuginfo_packages:
+        command_list.append(["sudo", "dnf", "-y", "debuginfo-install"] + debuginfo_packages)
+
+    retry = 0
+    while retry < 2:
+        for command in command_list:
+            try:
+                display(args, run_command(command), 0)
+            except RuntimeError as error:
+                # Abort running the list of commands if an error is encountered
+                display(args, error, 0)
+                retry += 1
+                break
+
+        if retry == 1:
+            # Re-run the lists of commands with a dnf clean and makecache
+            display(args, "Going to refresh caches and try again", 0)
+            command_prefix = ["sudo", "dnf"]
+            if is_el(distro_info) or "suse" in distro_info.name.lower():
+                command_prefix.append("--enablerepo=*debug*")
```

Comment:
This is `dnf debuginfo-install` _voodoo_ that we should not be trying to re-invent.  The repo reorg that is included in the Artifactory PR landing attempts to alleviate the need for this kind of _voodoo-reinvention_ so that `dnf debuginfo` should be able to do **all** of the heavy lifting here.

The Artifactory PR is in review and you (@phender) are one of the reviewers on it.  We should let that land and then see what kind of wheel-reinvention we can eliminate here.